internal/unionstore/art/art_snapshot.go

```diff
@@ -31,8 +31,8 @@ func (t *ART) getSnapshot() arena.MemDBCheckpoint {
 }
 
 // SnapshotGetter returns a Getter for a snapshot of MemBuffer.
-func (t *ART) SnapshotGetter() *SnapGetter {
-	return &SnapGetter{
+func (t *ART) SnapshotGetter() *Snapshot {
```

Comment:
Is it possible to keep only `Snapshot` function? As the upper callers should use the new `GetSnapshot` interface.

Reply:
Besides there are quite some `Snapshot` like definitions in the code.

Reply:
I remove the `SnapGetter` because it's a subset of `Snapshot`. Also make the `MemBuffer.SnapshotGetter` deprecated.

The current code doesn't have `SnapGetter` but there are still `uSnapshot` and `MemBufferSnapshot`, and some other `Snapshot` named structs for implementation. To make it more clear, I move the `SnapshotWithMutex` into an individual file.